lib/kernel/src/file.erl

```diff
@@ -398,24 +398,32 @@ write_file(Name, Bin) ->
       Reason :: posix() | badarg | terminated | system_limit.
 
 write_file(Name, Bin, ModeList) when is_list(ModeList) ->
-    case make_binary(Bin) of
-	B when is_binary(B) ->
-	    case open(Name, [binary, write | 
-			     lists:delete(binary, 
-					  lists:delete(write, ModeList))]) of
-		{ok, Handle} ->
-		    case write(Handle, B) of
-			ok ->
-			    close(Handle);
-			E1 ->
-			    _ = close(Handle),
-			    E1
-		    end;
-		E2 ->
-		    E2
-	    end;
-	E3 ->
-	    E3
+  case lists:member(raw, ModeList) of
+    true ->
+        do_write_file(Name, Bin, ModeList);
+    false ->
+        case make_binary(Bin) of
+            B when is_binary(B) ->
+                do_write_file(Name, B, ModeList);
+            Error ->
+                Error
+        end
+  end.
+
+do_write_file(Name, B, ModeList) ->
```

Comment:
@fenollp Actually, on second thought, I'm not whether a spec here makes sense. `do_write_file/3` is a private implementation detail of `write_file/3`; if it seems better, it could be refactored to be part of `write_file/3` with no effect on callers. As such, I think it should be covered by any tests of `write_file/3`.

If additional testing seems good, we might test the following: "if given an iolist and a mode list including `raw`, `write_file/3` will call `write` with its iolist, not with a binary". But I don't know how to write that test. I'm not even sure where the test would go - I'm guessing `read_write_file/1` in `lib/kernel/test/file_SUITE.erl`, but then it appears to me that the existing tests there are using a `write_file/2` that's defined at the bottom of that file, not the one touched by this PR.